iothub_client/samples/iothub_client_device_cbor_sample/iothub_client_device_cbor_sample.c

```diff
@@ -0,0 +1,956 @@
+            }
+        }
+        else
+        {
+            (void)printf("`%s` property name was not found in desired property message.\n", twin_property_state_max_speed_name);
+            result = false;
+        }
+
+        // state.software_version
+        rc = cbor_value_map_find_value(&state_root, twin_property_state_software_version_name, &software_version);
+        if (rc)
+        {
+            (void)printf("Error when searching for %s: CborError %d.\n", twin_property_state_software_version_name, rc);
+            exit(rc);
+        }
+        if (cbor_value_is_valid(&software_version))
+        {
+            if (cbor_value_is_double(&software_version))
+            {
+                rc = cbor_value_get_double(&software_version, &out_car->state.software_version);
+                if (rc)
+                {
+                    (void)printf("Failed to get double value of property %s: CborError %d.\n", twin_property_state_software_version_name, rc);
+                    exit(rc);
+                }
+                else
+                {
+                    (void)printf("Parsed desired property `%s`: %.*f\n", twin_property_state_software_version_name, 1, out_car->state.software_version);
+                    result = true;
+                }
+            }
+            else
+            {
+                (void)printf("`%s` property value was not a double.\n", twin_property_state_software_version_name);
+                result = false;
+            }
+        }
+        else
+        {
+            (void)printf("`%s` property name was not found in desired property message.\n", twin_property_state_software_version_name);
+            result = false;
+        }
+    }
+    else
+    {
+        (void)printf("`%s` property name was not found in desired property message.\n", twin_property_state_name);
+        result = false;
+    }
+
+    return result;
+}
+
+static void build_cbor_reported_property(uint8_t* reported_property_payload, size_t reported_property_payload_size, size_t* out_reported_property_payload_length)
+{
+    CborError rc; // CborNoError == 0
+
+    CborEncoder encoder;
+    CborEncoder encoder_map;
+    CborEncoder manufacturer_map;
+    CborEncoder state_map;
+
+    cbor_encoder_init(&encoder, reported_property_payload, reported_property_payload_size, 0);
+
+    // Encoder Map
+    rc = cbor_encoder_create_map(&encoder, &encoder_map, 4);
+    if (rc)
+    {
+        (void)printf("Failed to create encoder map: CborError %d.\n", rc);
+        exit(rc);
+    }
+
+        // change_oil_reminder
+        rc = cbor_encode_text_string(&encoder_map, twin_property_change_oil_reminder_name, strlen(twin_property_change_oil_reminder_name));
+        if (rc)
+        {
+            (void)printf("Failed to encode text string '%s': CborError %d.\n", twin_property_change_oil_reminder_name, rc);
+            exit(rc);
+        }
+        rc = cbor_encode_boolean(&encoder_map, device_car.change_oil_reminder);
```

Comment:
please add a space between previous `if()` blocks and new assignments
```suggestion
        }
        
        rc = cbor_encode_boolean(&encoder_map, device_car.change_oil_reminder);
```

Reply:
here and elsewhere below. find/all replace

Reply:
I'd like to leave as is because I think it's easier to read in groups.  If more people really don't like this I'll change it.